reladomographql/src/main/java/com/gs/reladomo/graphql/SchemaGenerator.java

```diff
@@ -0,0 +1,314 @@
+        {
+            String name = controller.getClassName ();
+            name = StringUtils.decapitalize (name, name.lastIndexOf ('.') + 1);
+            name = StringUtils.capitalize (name);
+            RelatedFinder finder = controller.getFinderInstance ();
+
+            out.write ("#################### " + controller.getClassName () + " ##################\n");
+
+            generateType (out, declaredTypes, name, finder);
+            generateAggregate (out, declaredTypes, name, finder);
+            generateInput (out, name, finder);
+            generateOrderBy (out, name, finder);
+            generateFinder (out, declaredTypes, name, finder);
+            generateTupleInput (out, name, finder);
+        }
+    }
+
+    private void generateType (Writer out, Set<String> declaredTypes, String name, RelatedFinder finder) throws IOException
+    {
+        out.write ("type " + name + "\n{\n");
+        for (Attribute attr : finder.getPersistentAttributes ())
+        {
+            this.writeAttribute (attr, out, (aName, type) -> "    " + aName + ": " + type + "\n");
+        }
+        if (finder.getAsOfAttributes () != null)
+        {
+            for (AsOfAttribute attr : finder.getAsOfAttributes ())
+            {
+                out.write ("    " + attr.getAttributeName () + ": Timestamp\n");
+            }
+        }
+
+        this.generateRelationshipBlock (out, finder, declaredTypes, false);
+        out.write ("}\n\n");
+    }
+
+    private void generateTupleInput (Writer out, String name, RelatedFinder finder) throws IOException
+    {
+        out.write ("input " + name + "FinderTuple\n{\n");
+        for (Attribute attr : finder.getPersistentAttributes ())
+        {
+            this.writeAttribute (attr, out, (aName, type) -> "    " + aName + ": " + type + "\n");
+        }
+
+//        this.writeRelationshipsBlock (out, finder, declaredTypes);
+        out.write ("}\n\n");
+    }
+
+
+    private void generateFinder (Writer out, Set<String> declaredTypes, String name, RelatedFinder finder) throws IOException
+    {
+        out.write ("input " + name + "Finder\n{\n");
+        for (Attribute attr : finder.getPersistentAttributes ())
+        {
+            this.writeAttribute (attr, out, (aName, type) -> {
+                String attrType = null;
+                switch (type)
+                {
+                    case "Int":
+                        attrType = "IntegerAttribute";
+                        break;
+                    case "Float":
+                        attrType = "FloatAttribute";
+                        break;
+                    case "String":
+                        attrType = "StringAttribute";
+                        break;
+                    case "Timestamp":
+                        attrType = "TimestampAttribute";
+                        break;
+                    case "Boolean":
+                        attrType = "BooleanAttribute";
+                        break;

```

Comment:
char?

Reply:
added handling of the CharAttribute. 
The graphql represents char values as a JSON string with one character.